indico/modules/events/reviewing_questions_fields.py

```diff
@@ -0,0 +1,103 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from wtforms.fields import BooleanField, StringField
+from wtforms.validators import DataRequired
+
+from indico.modules.events.fields import RatingReviewField
+from indico.util.i18n import _
+from indico.web.fields.base import BaseField, IndicoForm
+from indico.web.fields.simple import BoolField, TextField
+from indico.web.forms.widgets import SwitchWidget
+
+
+class BaseReviewingQuestionConfigForm(IndicoForm):
+    text = StringField(_('Question'), [DataRequired()])
+    is_required = BooleanField(_('Required'), widget=SwitchWidget())
+
+    @property
+    def field_data(self):
+        return {'is_required': self.is_required.data}
+
+
+class AbstractRatingReviewingQuestionConfigForm(BaseReviewingQuestionConfigForm):
+    no_score = BooleanField(_('Exclude from score'), widget=SwitchWidget())
+
+
+class PaperRatingReviewingQuestionConfigForm(BaseReviewingQuestionConfigForm):
+    pass
+
+
+class AbstractRatingReviewingQuestion(BaseField):
+    name = 'rating'
+    friendly_name = _('Rating')
+    common_settings = ('text', 'no_score')
+    config_form_base = AbstractRatingReviewingQuestionConfigForm
+    wtf_field_class = RatingReviewField
+
+    @property
+    def wtf_field_kwargs(self):
+        range_ = self.object.event.cfa.rating_range
+        choices = [(n, unicode(n)) for n in range(range_[0], range_[1] + 1)]
+        return {'coerce': int, 'choices': choices, 'rating_range': range_, 'question': self.object}
+
+
+class PaperRatingReviewingQuestion(BaseField):
+    name = 'rating'
+    friendly_name = _('Rating')
+    common_settings = ('text', 'no_score')
+    config_form_base = PaperRatingReviewingQuestionConfigForm
+    wtf_field_class = RatingReviewField
+
+    @property
+    def wtf_field_kwargs(self):
+        range_ = self.object.event.cfp.rating_range
+        choices = [(n, unicode(n)) for n in range(range_[0], range_[1] + 1)]
+        return {'coerce': int, 'choices': choices, 'rating_range': range_, 'question': self.object}
+
+
+class BoolReviewingQuestionConfigForm(BaseReviewingQuestionConfigForm):
+    pass
+
+
+class BoolReviewingQuestion(BoolField, BaseField):
+    common_settings = ('text',)
+    config_form_base = BoolReviewingQuestionConfigForm
+
+
+class TextReviewingQuestionConfigForm(BaseReviewingQuestionConfigForm):
+    _order = ('text', 'is_required', 'max_length', 'max_words', 'multiline')
+
+    @property
+    def field_data(self):
+        base = super(TextReviewingQuestionConfigForm, self).field_data
+        return dict(base, multiline=self.multiline.data, max_words=self.max_words.data,
+                    max_length=self.max_length.data)
+
+
+class TextReviewingQuestion(TextField, BaseField):
+    common_settings = ('text',)
+    config_form_base = TextReviewingQuestionConfigForm
+
+
+def get_reviewing_question_types():
```

Comment:
Why not add an argument for the type (abstracts/papers) instead of returning both?